packages/core/src/storages/request_queue_v2.ts

```diff
@@ -0,0 +1,277 @@
+import type { Dictionary } from '@crawlee/types';
+
+import type { RequestQueueOperationInfo, RequestProviderOptions } from './request_provider';
+import { RequestProvider } from './request_provider';
+import {
+    STORAGE_CONSISTENCY_DELAY_MILLIS,
+    getRequestId,
+} from './utils';
+import { Configuration } from '../configuration';
+import type { Request } from '../request';
+
+// Double the limit of RequestQueue v1 (1_000_000) as we also store keyed by request.id, not just from uniqueKey
+const MAX_CACHED_REQUESTS = 2_000_000;
+
+/**
+ * When prolonging a lock, we do it for 3 minutes from Date.now()
+ */
+const PROLONG_LOCK_BY_SECS = 3 * 60;
```

Comment:
Why do we have static time to lock? It should be based on requestHandlerTimeoutSecs + some reasonable buffer.
Imagine the requestHandlerTimeoutSecs will be more than 180 secs, then you cannot ensure that the request will not be picked by another parallel scraper.